pkg/virt-launcher/virtwrap/network/bindings_test.go

```diff
@@ -0,0 +1,164 @@
+package network
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vishvananda/netlink"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/cache/fake"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("BindingMechanism", func() {
+	testRunPlug := func(driver BindMechanism, podInterfaceName string) {
+		err := driver.discoverPodNetworkInterface(podInterfaceName)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(driver.preparePodNetworkInterface()).To(Succeed())
+		Expect(driver.decorateConfig(driver.generateDomainIfaceSpec())).To(Succeed())
+	}
+	Context("when slirp binding mechanism is selected", func() {
+		var (
+			domain                      *api.Domain
+			driver                      SlirpBindMechanism
+			vmi                         *kubevirtv1.VirtualMachineInstance
+			NewDomainWithSlirpInterface = func() *api.Domain {
```

Comment:
function defined within a variable block. and exported.

Reply:
[ditto](https://github.com/kubevirt/kubevirt/pull/5747#discussion_r645373478)